unlock-app/app/frames/checkout/txdata/route.ts

```diff
@@ -12,30 +10,14 @@ export const POST = frames(async (ctx) => {
     throw new Error('Invalid frame message')
   }
 
-  const userAddress = ctx.message.connectedAddress!
+  const userAddress = ctx.message.address!
+  const { address: lockAddress, priceForUser } = ctx.state.lock!
   const network = Number(ctx.state.lock!.network)
-  const lockAddress = ctx.state.lock!.address
-
-  async function getKeyPrice() {
-    const web3Service = new Web3Service(networks)
-    const mydata = '0x'
-    let price = await web3Service.purchasePriceFor({
-      lockAddress,
-      userAddress: userAddress,
-      network,
-      data: mydata,
-      referrer: userAddress,
-    })
-    price = price.toString()
-    return price
-  }
-
-  const keyPrice = await getKeyPrice()
 
   const calldata = encodeFunctionData({
```

Comment:
Please use unlock'js walletService here as well so this supports any version of the lock contract!